libs/agno/agno/agent/agent.py

```diff
@@ -4146,12 +4183,19 @@ def search_knowledge_base(self, query: str) -> str:
         Returns:
             str: A string containing the response from the knowledge base.
         """
+        # Get knowledge filters from context if available
+        filters = None
+        if self.context and "knowledge_filters" in self.context:
```

Comment:
Why? Lets just pass it along from the run. Context is used in the system prompt to the models.